app/ante/vesting.go

```diff
@@ -67,31 +86,79 @@ func (vtd EthVestingTransactionDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx,
 		// Check that this decorator only applies to clawback vesting accounts
 		clawbackAccount, isClawback := acc.(*vestingtypes.ClawbackVestingAccount)
 		if !isClawback {
-			return next(ctx, tx, simulate)
+			continue
 		}
 
-		// Error if vesting cliff has not passed (with zero vested coins). This
-		// rule does not apply for existing clawback accounts that receive a new
-		// grant while there are already vested coins on the account.
-		vested := clawbackAccount.GetVestedCoins(ctx.BlockTime())
-		if len(vested) == 0 {
-			return ctx, errorsmod.Wrapf(vestingtypes.ErrInsufficientVestedCoins,
-				"cannot perform Ethereum tx with clawback vesting account, that has no vested coins: %s", vested,
-			)
+		// Check to make sure that the account does not exceed its spendable balances.
+		// This transaction would fail in processing, so we should prevent it from
+		// moving past the AnteHandler.
+		msgValue := msgEthTx.AsTransaction().Value()
+
+		totalSpend, err := vtd.getTotalSpend(ctx, totalSpendByAddress, clawbackAccount, msgValue, denom)
+		if err != nil {
+			return ctx, err
 		}
 
-		// Error if account has locked coins (before surpassing all lockup periods)
-		islocked := clawbackAccount.HasLockedCoins(ctx.BlockTime())
-		if islocked {
-			return ctx, errorsmod.Wrapf(vestingtypes.ErrVestingLockup,
-				"cannot perform Ethereum tx with clawback vesting account, that has locked coins: %s", vested,
+		totalValue := totalSpend.totalValue
+		spendableValue := totalSpend.spendableValue
+
+		if totalValue.Cmp(spendableValue) > 0 {
+			return ctx, errorsmod.Wrapf(vestingtypes.ErrInsufficientUnlockedCoins,
+				"clawback vesting account has insufficient unlocked tokens to execute transaction: %s < %s", spendableValue.String(), totalValue.String(),
 			)
 		}
 	}
 
 	return next(ctx, tx, simulate)
 }
 
+// getTotalSpend updates or sets the totalSpend for the given account, then
+// returns the new value.
+func (vtd EthVestingTransactionDecorator) getTotalSpend(
+	ctx sdk.Context,
+	totalSpendByAddress map[string]*ethVestingTotalSpend,
+	account *vestingtypes.ClawbackVestingAccount,
+	value *big.Int,
+	denom string,
+) (*ethVestingTotalSpend, error) {
+	address := account.GetAddress()
+
+	totalSpend, ok := totalSpendByAddress[address.String()]
+	if ok {
+		totalSpend.totalValue.Add(totalSpend.totalValue, value)
```

Comment:
I don't see where new value being updated in the map?

Reply:
ah it's using the pointer. I think it's better to use it as follow 
```suggestion
		totalSpend.totalValue = totalSpend.totalValue.Add(totalSpend.totalValue, value)
		totalSpendByAddress[address.String()] = totalSpend
```